strawberry_django/type.py

```diff
@@ -79,9 +80,29 @@ def _process_type(
     select_related: Optional[TypeOrSequence[str]] = None,
     prefetch_related: Optional[TypeOrSequence[PrefetchType]] = None,
     disable_optimization: bool = False,
+    fields: Optional[Union[List[str], str]] = None,
+    exclude: Optional[List[str]] = None,
     **kwargs,
 ) -> _O:
     is_input = kwargs.get("is_input", False)
+
+    if not exclude:
+        exclude = []
+
+    if fields == "__all__":
+        model_fields = list(model._meta.fields)
+    elif isinstance(fields, collections.abc.Sequence):
+        model_fields = [f for f in model._meta.fields if f.name in fields]
+    elif isinstance(exclude, collections.abc.Sequence) and len(exclude) > 0:
+        model_fields = [f for f in model._meta.fields if f.name not in exclude]
+    else:
+        model_fields = []
+
+    for f in model_fields:
+        if cls.__annotations__.get(f.name):
```

Comment:
You need to use `get_annotations` here to also consider annotations from parents

```suggestion
    existing_annotations = set(get_annotations(cls).keys())
    for f in model_fields:
        if f.name in existing_annotations:
```